src/core/namespace.c

```diff
@@ -519,6 +576,14 @@ static int append_extensions(
                                         "No matching entry in .v/ directory %s found.",
                                         m->source);
 
+                /* If this is a dm-verity image, check for duplicates, as Overlayfs rejects them as
+                 * explained above. */
+                r = verities_find(result.path, &verities, &n_verities);
+                if (r < 0)
+                        return r;
+                if (r == 0) /* This is a duplicate, skip it */
+                        continue;
```

Comment:
hmm, I am pretty sure this is actually the wrong place for this. i.e. we shouldn't filter when putting this together, but when we actually process this stuff. i.e. in mount_image() and/or verity_dissect_and_mount().

i.e. it really sucks loading the stuff ahead of time, then flushing it out, and reading it again later. That's just sloppy.

i.e. maybe update  verity_dissect_and_mount() to take the VeritySettings to use already as input. Then load the stuff right before calling that, and filter duplicates.

Reply:
That's too late, as the overlay lowerdir= is already compiled, that's why it needs to happen here. You would need to detect something was skipped, go back and monkey patch ALL the overlay lowerdirs, for each hierarchy, to drop the deduped path.

Reply:
Unless maybe you mean that the mountpoint would simply be left empty, but still passed to lowerdir= ? So only the image mount is dropped, but the parameters passed to the overlay mount remain intact, and the duplicate is simply an unmounted, empty mount point? I guess that would be doable, if that is what you mean.

Still feels weird to me, as the entire setup here is done intentionally to _first_ check everything, read all the info without doing any changes anywhere, adjust/deduplicate/modify/rearrange the intended state (which is what this PR does), and then later, separately, apply the changes so the intended state is reached. Changing this means deduplication happens in the second stage.

Reply:
> That's too late, as the overlay lowerdir= is already compiled, that's why it needs to happen here. You would need to detect something was skipped, go back and monkey patch ALL the overlay lowerdirs, for each hierarchy, to drop the deduped path.

No, just maintain the lowerdir thing an strv, and compile it the moment you mount the overlayfs, and not ahead of time. Then you can make changes still.

or actually, we process the mount list heavily anyway, and suppress stuff that is obstructed. Hence maybe don't bother with an strv either, but at the moment where the overlayfs is to be applied, simply just iterate throught the list of mounts actually established and see what actually go applied, and filter by a mount path prefix match or so to find the right ones. That way, we can freely drop entries for *any* reason *any* time, and be sure the right thing happens.